# Evaluation

This note book provides anlytics and comparison of the created "esoc_all_data_analytics.xlsx" and the "75_12_Master_v4rc03_full.xlsx" document provided in the research documents.

## Cleaning the master to fit the analytics

This code simply modifies the master list to fit the date range and column names of the analytics

In [96]:
import pandas as pd
import numpy as np
import os
#First cut and make new csv with only year, this is done first so large xslx does not need to be reloaded later
esoc_master_raw = pd.read_excel("esoc_master_raw.xlsx")
esoc_master_raw_2010_2012 = esoc_master_raw.loc[(esoc_master_raw['year'] == 2010) | (esoc_master_raw['year'] == 2011) | (esoc_master_raw['year'] == 2012)]

#Print number of entries in this range for later verification
print esoc_master_raw_2010_2012.shape

#Print to csv
esoc_master_raw_2010_2012.to_csv("Master_2010_2012.csv",header=True, index=False, encoding='utf-8')

(4497, 75)


In [97]:
#Get cut data from 2010-2012
esoc_master = pd.read_csv("Master_2010_2012.csv")

#Duplicated Journal Numbers in the master file
#jxid has no duplicates, but its origins are unknown
master_duplicates = pd.concat(g for _, g in esoc_master.groupby("jnmbr") if len(g) > 1)
print len(master_duplicates)

#Remove duplicates for comparison adn then put numbers into series
master_unique = esoc_master.drop_duplicates("jnmbr")
master_journal_nums = master_unique["jnmbr"]
print len(master_journal_nums)

27
4483


In [98]:
#Get data from In-Processing and then drop possible duplicates 
#Only one seen in trials
esoc_all_data_raw = pd.read_csv("ESOC_All_Data_2010_2012_Analytics.csv")
raw_duplicates = pd.concat(g for _, g in esoc_all_data_raw.groupby("Journal Code") if len(g) > 1)
print len(raw_duplicates)

#Add uniques to series
raw_unique = esoc_all_data_raw.drop_duplicates("Journal Code")
raw_journal_nums = raw_unique["Journal Code"]
print len(raw_journal_nums)


6
9838


In [106]:
#Find number of matching journal numbers between raw and master and then put them into a list
mathcing_journal_nums = pd.Series(list(set(master_journal_nums) & set(raw_journal_nums)))
print len(mathcing_journal_nums)

#Make new dataframes with only the matched journals for companrison and sort by, and set as index
raw_matched_journals = raw_unique[raw_unique['Journal Code'].isin(mathcing_journal_nums)]
master_matched_journals = master_unique[master_unique["jnmbr"].isin(mathcing_journal_nums)]

print len(raw_matched_journals)
print len(master_matched_journals)

3408
3408
3408


In [107]:
#Set Journal Number as the Index and sort
#Make new dataframes with only the matched journals for companrison and sort by, and set as index
raw_matched_journals = raw_matched_journals.set_index("Journal Code")
master_matched_journals = master_matched_journals.set_index("jnmbr")

In [109]:
pd.concat([master_matched_journals["govtwia"],raw_matched_journals["GS WIA"],raw_matched_journals["Date"]], axis=1)

govtwia  GS WIA        Date
10001           0       0  2010-06-03
10005           1       1  2010-06-04
10007           0       2  2010-06-04
10009           0       0  2010-06-05
10012           0       0  2010-06-05
10013           0       0  2010-06-05
10015           0       0  2010-06-05
10017           0       0  2010-06-05
10019           0       0  2010-06-06
10021           0       0  2010-06-05
10024           1       1  2010-06-05
10026           0       0  2010-06-06
10032           0       0  2010-06-06
10033           5       6  2010-06-07
10034           0       0  2010-06-06
10046           0       0  2010-06-07
10047           0       0  2010-06-07
10051           0       0  2010-06-08
10053           0       0  2010-06-08
10054           0       0  2010-06-08
10058           1       1  2010-06-08
10059           0       0  2010-06-08
10071           0       0  2010-06-10
10072           0       0  2010-06-10
10073           0       0  2010-06-10
10078           0       0  2010-06-11
10081           0       0  2010-06-11
10082           0       0  2010-06-11
10085           0       0  2010-06-11
10086           0       0  2010-06-11
...           ...     ...         ...
EMC12738        0       0  2012-12-12
EMC12740        0       0  2012-12-12
EMC12744        0       0  2012-12-13
EMC12753        0       0  2012-12-16
EMC12759        0       0  2012-12-18
EMC12760        0       0  2012-12-18
EMC12773        0       0  2012-12-22
EMC12777        0       0  2012-12-25
EMC12778        0       0  2012-12-25
EMC12781        0       0  2012-12-27
EMC12790        0       0  2012-12-27
EMC13748        0       0  2012-12-13
EMC16311        0       0  2012-10-16
WMC10363        0       0  2010-07-08
WMC10364        0       0  2010-07-07
WMC10366        0       0  2010-06-29
emc02019        1       1  2012-02-05
emc05840        0       0  2012-05-30
emc08212        0       0  2012-08-24
emc08221        0       0  2012-08-24
emc08638        0       0  2011-08-26
emc10241        0       0  2012-10-06
emc10368        0       0  2012-10-22
emc10404        1       0  2012-10-29
emc11041        0       0  2011-11-20
emc11625        0       0  2012-11-28
emc11629        0       0  2012-11-28
emc12144        0       0  2011-12-05
emc12720        0       0  2012-12-10
emc12783        0       0  2012-12-26

[3408 rows x 3 columns]

After review of this data the master list seems to have an error for Journal Numner 10007: 
<img src="journal_10007.png">